Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -2074,6 +2074,17 @@ ImageOrientation RenderElement::imageOrientation() const
     return (imageElement && !imageElement->allowsOrientationOverride()) ? ImageOrientation(ImageOrientation::Orientation::FromImage) : style().imageOrientation();
 }
 
+#if ENABLE(MULTI_REPRESENTATION_HEIC)
+bool RenderElement::isMultiRepresentationHEIC() const
+{
+    RefPtr imageElement = dynamicDowncast<HTMLImageElement>(element());
+    if (!imageElement)
+        return false;
+
+    return imageElement->isMultiRepresentationHEIC();
+}
```

Comment:
any particular reason why this is not on RenderImage (e.g. this is going to evolve to a more generic approach)?

Reply:
Not going to evolve to a more generic approach.

I have it here because it's the same way `imageOrientation` is done above, and we have a `RenderElement *` at the call site. Was just hoping to avoid introducing new checks + a virtual method.

Reply:
there's only one call site outside of RenderImage afaict. I don't think it's a good approach to move all the things to the base class and downcast there to specific types (and I don't see why it would require to introduce a virtual method)

Reply:
Good point – will make that change.